From faad748af24b9bbbaa894243c9cf52fd841bb8d4 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Tue, 20 Aug 2013 19:37:29 +0100
Subject: [PATCH 28/56] TV: Quit the TV playback loop when the primary player
 reports a fatal error

If MythPlayer::OpenFile fails (for instance network congestion) then
TV::StartTV continues by restarting TV instance which then causes problems
for the backend.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
Signed-off-by: Jean-Yves Avenard <jyavenard@mythtv.org>
---
 mythtv/libs/libmythtv/tv_play.cpp |    1 +
 1 file changed, 1 insertion(+)

diff --git a/mythtv/libs/libmythtv/tv_play.cpp b/mythtv/libs/libmythtv/tv_play.cpp
index 213a80e..fdde7ae 100644
--- a/mythtv/libs/libmythtv/tv_play.cpp
+++ b/mythtv/libs/libmythtv/tv_play.cpp
@@ -434,6 +434,7 @@ bool TV::StartTV(ProgramInfo *tvrec, uint flags,
             mctx->UnlockDeletePlayer(__FILE__, __LINE__);
         }
         tv->ReturnPlayerLock(mctx);
+        quitAll |= !playerError.isEmpty();
     }
 
     LOG(VB_PLAYBACK, LOG_INFO, LOC + "StartTV -- process events 2 begin");
-- 
1.7.10.2

